app/scripts/modules/core/src/pipeline/manualExecution/ManualPipelineExecutionModal.tsx

```diff
@@ -0,0 +1,414 @@
+import { Registry } from 'core/registry';
+import { SETTINGS } from 'core/config/settings';
+import { UrlParser } from 'core/navigation/urlParser';
+
+import { PipelineOptions } from './PipelineOptions';
+import { CurrentlyRunningExecutions } from './CurrentlyRunningExecutions';
+import { StageManualComponents } from './StageManualComponents';
+import { Triggers } from './Triggers';
+import { Parameters } from './Parameters';
+import { DryRun } from './DryRun';
+import { NotificationDetails } from './NotificationDetails';
+
+import './manualPipelineExecution.less';
+
+export interface IManualExecutionModalProps extends IModalComponentProps {
+  application: Application;
+  currentlyRunningExecutions?: IExecution[];
+  pipeline?: IPipeline;
+  trigger?: IExecutionTrigger;
+}
+
+export interface IManualExecutionModalState {
+  applicationNotifications: INotification[];
+  currentPipelineExecutions: IExecution[];
+  dryRunEnabled?: boolean;
+  modalHeader: string;
+  pipelineNotifications: INotification[];
+  pipelineOptions?: IPipeline[];
+  stageComponents: Array<React.ComponentType<ITriggerTemplateComponentProps>>;
+  triggerComponent?: React.ComponentType<ITriggerTemplateComponentProps>;
+  triggers: ITrigger[];
+}
+
+export class ManualExecutionModal extends React.Component<IManualExecutionModalProps, IManualExecutionModalState> {
+  private destroy$ = new Subject();
+
+  constructor(props: IManualExecutionModalProps) {
+    super(props);
+    let modalHeader: string;
+    if (props.pipeline) {
+      modalHeader = 'Select Pipeline';
+      if (props.pipeline.triggers) {
+        modalHeader = 'Select Execution Parameters';
+      } else {
+        modalHeader = 'Confirm Execution';
+      }
+    } else {
+      modalHeader = 'Select Pipeline';
+    }
+    this.state = {
+      applicationNotifications: [],
+      currentPipelineExecutions: props.currentlyRunningExecutions || [],
+      modalHeader,
+      pipelineNotifications: [],
+      pipelineOptions: [],
+      stageComponents: [],
+      triggers: [],
+    };
+  }
+
+  public componentDidMount() {
+    const { application, pipeline, trigger } = this.props;
+    let pipelineOptions = [];
+    let pipelineNotifications: INotification[] = [];
+    let triggers: ITrigger[] = [];
+    if (pipeline) {
+      pipelineNotifications = pipeline.notifications || [];
+      if (pipeline.triggers) {
+        triggers = this.formatTriggers(pipeline.triggers);
+      }
+    } else {
+      pipelineOptions = application.pipelineConfigs.data.filter(
+        (c: any) => !c.disabled && PipelineTemplateV2Service.isConfigurable(c),
+      );
+    }
+
+    this.triggerChanged(trigger);
+
+    Observable.fromPromise(AppNotificationsService.getNotificationsForApplication(application.name))
+      .takeUntil(this.destroy$)
+      .subscribe(notifications => {
+        const applicationNotifications: INotification[] = [];
```

Comment:
it shouldn't be, I lifted the logic from the previous implementation but I don't see how it can be anything different than `undefined` or an array of `INotification`

Reply:
Yeah, sorry about that.  It can be hard to distinguish between what code was lifted from the old code during a refactor!